h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -71,6 +73,28 @@ public void init(boolean expensive) {
     super.init(expensive);
 
     checkFoldColumnPresent(_parms._metalearner_fold_column, train(), valid(), _parms.blending());
+    validateAndExpandBaseModels();
+  }
+
+  /**
+   * Validates base models and if grid is provided instead of a model it gets expanded.
+   */
+  private void validateAndExpandBaseModels() {
+    List<Key> base_models = new ArrayList<Key>();
+    for (Key base_model: _parms._base_models) {
+      Object retrieved_object = DKV.getGet(base_model);
+      if (retrieved_object instanceof Model) {
+        base_models.add(base_model);
+      } else if (retrieved_object instanceof Grid) {
+        Grid grid = (Grid) retrieved_object;
+        Collections.addAll(base_models, grid.getModelKeys());
```

Comment:
as you're using an ArrayList, I don't think you get any benefit in using `Collections.addAll`, perf is probably even worse than a simple `list.addAll(...)`

Reply:
Maybe I am missing something, but I see 3 options I have here - `getModelKeys` returns `Key<Model>[]`.

So either I use:
1. `Collections.addAll(baseModels, grid.getModelKeys());`, or
2. `baseModels.addAll(Arrays.asList(grid.getModelKeys()));`, or
3. for cycle adding one element at a time

Or is there some better way of doing it?

Reply:
are you concerned about the generic type? solutions #1 and #2 will bring an additional cost, just to be able to sweep the explicit cast under the rug.
Explicit cast should work:
```
baseModels.addAll((Key[])grid.getModelKeys());
```

Reply:
sorry, I though that `grid.getModelKeys()` was returning  a `List`